### PR TITLE
Add download functionality

### PR DESCRIPTION
**Pull request by DenisaCG**
Add `download` functionality for all files. The `@aws-sdk/s3-request-presigner` library was added to help generate a pre-signed URL for each `S3` stored file through which the user can download the file locally. 

For certain media type files, such as `pdf`, `png`, `jpg` or `svg`, the files are stored in the `S3` bucket encoded as `utf-8` arrays and converted to `base64` when viewed in `JupyterLab`. This logic allows for the correct storing and downloading of files from the bucket, as well as the files having no formatting issues in `JupyterLab`. The `save` and `get` functionalities have been updated accordingly.

Additionally, all registered file types get extracted dynamically, by checking the `docRegistry`. This ensures file types coming from other extensions the user has installed, on top of the normal `JupyterLab` ones, will be supported.

### CODE REVIEW

File: src/s3contents.ts [hunk 1]
```diff
@@ -176,9 +205,21 @@ export class Drive implements Contents.IDrive {
    * use [[ContentsManager.getAbsolutePath]] to get an absolute
    * path if necessary.
    */
-  getDownloadUrl(path: string): Promise<string> {
-    // Parse the path into user/repo/path
-    return Promise.reject('Empty getDownloadUrl method');
+  async getDownloadUrl(path: string): Promise<string> {
+    console.log('GET DOWNLOAD URL, given path: ', path);
+
```
Comment (afshin) [inline src/s3contents.ts:210]:
```suggestion
```

File: src/s3contents.ts [hunk 2]
```diff
@@ -1138,131 +1212,61 @@ export class Drive implements Contents.IDrive {
...
-        fileFormat = 'json';
-        break;
-      default:
-        fileType = 'text';
-        fileMimetype = 'text/plain';
-        break;
+    let fileType: string = 'text';
+    let fileMimetype: string = 'text/plain';
+    let fileFormat: string = 'text';
+    extension = extension !== undefined ? extension : '';
```
Comment (afshin) [inline src/s3contents.ts:1259]:
```suggestion
    extension = extension ?? '';
```

**Review by afshin — APPROVED** [review]

**Comment by afshin** [comment]
Thank you!

**Comment by DenisaCG** [comment]
Thanks for the review @afshin!

**Pull request merged**